src/common/ui/ProgressWindow.jsx

```diff
@@ -261,6 +289,50 @@ Zotero.UI.ProgressWindow = class ProgressWindow extends React.PureComponent {
 		});
 	}
 	
+	/**
+	 * Announce an alert for screen readers. To handle multiple events coming in at the same time,
+	 * messages go into a queue from where processMessageQueue() polls and adds them to the alert note
+	 * If there are multiple events with the same id being added to the queue at the 
+	 * same time (e.g. download started and download finished) - only the last event is kept.
+	 * @param {string} text - message to be announced
+	 * @param {string} id - id of the item for deduplication
+	 */
+	setAlertMessage(text, id) {
+		let alertNode = document.getElementById("messageAlert");
+
+		this.alertQueue = this.alertQueue.filter(obj => obj.id !== id);
+
+		this.alertQueue.push({message: text, id: id });
+
+		// Initialize the loop that processes the alert queue
+		if (alertNode.textContent.length == 0) {
+			this.processMessageQueue();
+		}
+	}
+
+	/**
+	 * Periodically poll the queue of alert messages and place them into the alert node.
+	 */
+	processMessageQueue = () => {
+		const internal = 2000;
+		let alertNode = document.getElementById("messageAlert");
+
+		// If the queue is empty, wait
+		if (this.alertQueue.length == 0) {
+			setTimeout(this.processMessageQueue, internal);
+			return;
+		}
+		// Get the first message and announce if it is not the same a the last alert (sanity check)
+		let { message, _ } = this.alertQueue[0];
+		if (alertNode.textContent != message) {
+			alertNode.textContent = message;
+		}
+		// Delete the element from the queue
+		this.alertQueue.shift();
+		// Wait a bit and then try to announce the next message
+		setTimeout(this.processMessageQueue, internal);
+	}
```

Comment:
There are clean ways to sequentialize/chain calls without infinite waiting loops. E.g. see https://github.com/zotero/zotero-connectors/blob/239745e644ae774192f8b390adbaaab30103a03c/src/common/inject/inject.jsx#L292-L315